python/pyspark/ml/torch/distributor.py

```diff
@@ -1003,3 +1006,142 @@ def _get_spark_partition_data_loader(
         # if num_workers is zero, we cannot set `prefetch_factor` otherwise
         # torch will raise error.
         return DataLoader(dataset, batch_size, num_workers=num_workers)
+
+
+class DeepspeedTorchDistributor(TorchDistributor):
+    
+    def __init__(self, num_gpus: int = 1, nnodes: int = 1, local_mode: bool = True, use_gpu: bool = True, deepspeed_config = None):
+        """
+            This class is used to run deepspeed training workloads with spark clusters. The user has the option to 
+            specify the number of gpus per node and the number of nodes (the same as if running from terminal), 
+            as well as specify a deepspeed configuration file.
+
+            Parameters
+            ----------
+            num_gpus: int
+                the number of GPUs to use per node (analagous to num_gpus in deepspeed command)
```

Comment:
Clean these up a bit please with capitalization and periods at the end.

Reply:
Done!